pkg/image/anaconda_tar_installer.go

```diff
@@ -94,26 +94,27 @@ func (img *AnacondaTarInstaller) InstantiateManifest(m *manifest.Manifest,
 	buildPipeline := addBuildBootstrapPipelines(m, runner, repos, nil)
 	buildPipeline.Checkpoint()
 
-	if img.Kickstart == nil {
-		img.Kickstart = &kickstart.Options{}
-	}
-
-	if img.Kickstart.Unattended {
+	if img.Kickstart != nil && img.Kickstart.Unattended {
 		// if we're building an unattended installer, override the
 		// ISORootKickstart option
 		img.ISORootKickstart = true
 	}
 
-	if img.ISORootKickstart {
+	if img.Kickstart != nil && img.ISORootKickstart {
 		// kickstart file will be in the iso root and not interactive-defaults,
 		// so let's make sure the kickstart path option is set
 		if img.Kickstart.Path == "" {
 			img.Kickstart.Path = osbuild.KickstartPathOSBuild
 		}
 	}
 
+	installerType := manifest.AnacondaInstallerTypeNetinst
+	if img.Kickstart != nil {
+		installerType = manifest.AnacondaInstallerTypePayload
+	}
```

Comment:
Why do we do this? We always switch to a payload installer when there's a kickstart?